crates/transaction-pool/src/validate.rs

```diff
@@ -80,6 +82,76 @@ pub trait TransactionValidator: Send + Sync {
     }
 }
 
+pub(crate) struct EthTransactionValidatorConfig<Client: AccountProvider> {
+    ///
+    chain_id: u64,
+    /// This type fetches account info from the db
+    client: Client,
+    /// whether shanghai is enabled.
+    shanghai: bool,
+    ///  
+    currentMaxGasLimit: u64,
+}
+
+#[async_trait::async_trait]
+impl<T: PoolTransaction + AccountProvider> TransactionValidator
+    for EthTransactionValidatorConfig<T>
+{
+    type Transaction = T;
+
+    async fn validate_transaction(
+        &self,
+        origin: TransactionOrigin,
+        transaction: Self::Transaction,
+    ) -> Result<TransactionValidationOutcome<Self::Transaction>, Error> {
```

Comment:
is this also missing the 128KB size check?

Reply:
Yes. Should I implement all the checks as in the [geth's validate function](https://github.com/ethereum/go-ethereum/blob/5d8bff1d7261aed1a68235b0223d44a4fb4fb259/core/txpool/txpool.go#L590)?

Reply:
yes that'd be ideal